jsr352/core/src/test/java/org/hibernate/search/jsr352/massindexing/MassIndexingJobParametersBuilderTest.java

```diff
@@ -14,78 +14,49 @@
 
 /**
  * @author Mincong Huang
  */
-@RunWith(MockitoJUnitRunner.class)
-public class BatchIndexingJobTest {
+public class MassIndexingJobParametersBuilderTest {
 
 	private static final String SESSION_FACTORY_NAME = "someUniqueString";
 
 	private static final boolean OPTIMIZE_AFTER_PURGE = true;
-	private static final boolean OPTIMIZE_AT_END = true;
-	private static final boolean PURGE_AT_START = true;
+	private static final boolean OPTIMIZE_ON_FINISH = true;
+	private static final boolean PURGE_ALL_ON_START = true;
 	private static final int FETCH_SIZE = 100000;
-	private static final int MAX_RESULTS = 1000000;
+	private static final int CUSTOM_QUERY_LIMIT = 1000000;
 	private static final int MAX_THREADS = 2;
 	private static final int ROWS_PER_PARTITION = 500;
 
-	@Mock
-	private JobOperator mockedOperator;
-
-	@Mock
-	private SessionFactoryImplementor mockedEMF;
-
-	@Before
-	public void setUp() {
-		Mockito.when( mockedOperator.start( Mockito.anyString(), Mockito.any( Properties.class ) ) )
-				.thenReturn( 1L );
-		Mockito.when( mockedEMF.isOpen() ).thenReturn( true );
-	}
-
 	@Test
 	public void testJobParamsAll() throws IOException {
-
-		ArgumentCaptor<Properties> propsCaptor = ArgumentCaptor.forClass( Properties.class );
-		long executionID = BatchIndexingJob.forEntities( String.class, Integer.class )
+		Properties props = MassIndexingJob.parameters()
+				.forEntities( String.class, Integer.class )
 				.entityManagerFactoryReference( SESSION_FACTORY_NAME )
-				.underJavaSE( mockedOperator )
 				.fetchSize( FETCH_SIZE )
-				.maxResults( MAX_RESULTS )
+				.customQueryLimit( CUSTOM_QUERY_LIMIT )
 				.maxThreads( MAX_THREADS )
 				.optimizeAfterPurge( OPTIMIZE_AFTER_PURGE )
-				.optimizeAtEnd( OPTIMIZE_AT_END )
+				.optimizeOnFinish( OPTIMIZE_ON_FINISH )
 				.rowsPerPartition( ROWS_PER_PARTITION )
-				.purgeAtStart( PURGE_AT_START )
-				.start();
-		assertEquals( 1L, executionID );
-
-		Mockito.verify( mockedOperator )
-				.start( Mockito.anyString(), propsCaptor.capture() );
-		Properties props = propsCaptor.getValue();
-		assertEquals( SESSION_FACTORY_NAME, props.getProperty( "entityManagerFactoryReference" ) );
-		assertEquals( FETCH_SIZE, Integer.parseInt( props.getProperty( "fetchSize" ) ) );
-		assertEquals( MAX_RESULTS, Integer.parseInt( props.getProperty( "maxResults" ) ) );
-		assertEquals( OPTIMIZE_AFTER_PURGE, Boolean.parseBoolean( props.getProperty( "optimizeAfterPurge" ) ) );
-		assertEquals( OPTIMIZE_AT_END, Boolean.parseBoolean( props.getProperty( "optimizeAtEnd" ) ) );
-		assertEquals( ROWS_PER_PARTITION, Integer.parseInt( props.getProperty( "rowsPerPartition" ) ) );
-		assertEquals( PURGE_AT_START, Boolean.parseBoolean( props.getProperty( "purgeAtStart" ) ) );
-		assertEquals( MAX_THREADS, Integer.parseInt( props.getProperty( "maxThreads" ) ) );
-
-		String rootEntities = propsCaptor.getValue().getProperty( "rootEntities" );
+				.purgeAllOnStart( PURGE_ALL_ON_START )
+				.build();
```

Comment:
Yes, you're right, parameters is not in the builder, but it's how we instantiate the builder. Let's keep `parameters`.